### PR TITLE
chore: upload lambda image cleanup

### PR DESCRIPTION
**Pull request by metakuni**
### Reviewers
**Functional:** 
@Bento007

**Readability:** 
@danieljhegeman

---

## Changes
- modify: This PR is less scary than it looks. 😄 Most of the files are modified to refactor the HTTP-specific exceptions into a separate module so that the upload lambda Docker image doesn't have to drag in portal API-related modules/packages.

## QA steps (optional)
* Deployed to rdev and ran a few sanity checks

### CODE REVIEW

File: Dockerfile.handle_dataset_submissions [hunk 1]
```diff
@@ -5,17 +5,6 @@ COPY backend/corpora/__init__.py ./backend/corpora/__init__.py
 COPY backend/corpora/common ./backend/corpora/common
 COPY backend/corpora/dataset_submissions .
 
-# TODO: remove after the authz refactoring (_owner_or_allowed)
```
Comment (metakuni) [inline Dockerfile.handle_dataset_submissions:8]: Buh bye!

File: backend/corpora/common/authorizer.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 from jose.exceptions import ExpiredSignatureError, JWTError, JWTClaimsError
 
 from .corpora_config import CorporaAuthConfig
-from backend.corpora.common.utils.exceptions import UnauthorizedError
+from backend.corpora.common.utils.http_exceptions import UnauthorizedError
```
Comment (metakuni) [inline backend/corpora/common/authorizer.py:9]: Moved HTTP-specific exceptions used by the portal API into a separate module.

File: backend/corpora/common/utils/exceptions.py [hunk 3]
```diff
@@ -1,7 +1,3 @@
-import requests
```
Comment (metakuni) [inline backend/corpora/common/utils/exceptions.py:1]: `requests` and `connexion.exceptions` are not needed by non-HTTP exceptions.

File: backend/corpora/common/utils/exceptions.py [hunk 4]
```diff
@@ -25,54 +21,3 @@ class InvalidProcessingStateException(CorporaException):
 
 class NonExistentDatasetException(CorporaException):
     pass
-
-
-class ColorFormatException(ProblemException):
```
Comment (metakuni) [inline backend/corpora/common/utils/exceptions.py:30]: Moved HTTP-specific exceptions to `backend.corpora.common.utils.http_exceptions`.

**Review by Bento007 — APPROVED** [review]
Nice refactor

**Comment by danieljhegeman** [comment]
Thanks @metakuni !!

**Pull request merged**